libcli.c

```diff
@@ -1861,37 +1900,22 @@ struct cli_range_filter_state {
 
 int cli_range_filter_init(struct cli_def *cli, int argc, char **argv, struct cli_filter *filt) {
   struct cli_range_filter_state *state;
-  char *from = 0;
-  char *to = 0;
-
-  if (!strncmp(argv[0], "bet", 3))  // between
-  {
-    if (argc < 3) {
-      if (cli->client) fprintf(cli->client, "Between filter requires 2 arguments\r\n");
-
-      return CLI_ERROR;
-    }
-
-    if (!(from = strdup(argv[1]))) return CLI_ERROR;
-    to = join_words(argc - 2, argv + 2);
-  } else  // begin
-  {
-    if (argc < 2) {
-      if (cli->client) fprintf(cli->client, "Begin filter requires an argument\r\n");
-
-      return CLI_ERROR;
-    }
-
-    from = join_words(argc - 1, argv + 1);
-  }
+  char *from = strdup(cli_get_optarg_value(cli, "range_start", NULL));
+  char *to = strdup(cli_get_optarg_value(cli, "range_end", NULL));
 
+  // Do not have to check from/to since we would not have gotten here if we were
+  // missing a required argument
   filt->filter = cli_range_filter;
   filt->data = state = calloc(sizeof(struct cli_range_filter_state), 1);
-
-  state->from = from;
-  state->to = to;
-
-  return CLI_OK;
+  if (state) {
```

Comment:
There's a check for the success of `calloc` but not `strdup`

Reply:
Yeah.  Our engineering lead on my main project has mostly had us not check mem allocs due to the opportunistic memory allocation stuff (overcommit) of modern Linux.  Granted the embedded world can be much different.  Good point, and after a quick check there are other malloc/calloc/strdup calls unchecked.